test/integration/targets/file/tasks/main.yml

```diff
@@ -62,6 +62,22 @@
       - "file4_result.changed == true"
       - "file4_result.mode == '0600'"
 
+- name: change file attribute "A"
+  file: path={{output_dir}}/baz.txt attributes=A
+  register: file_attributes_result
+  ignore_errors: True
+
```

Comment:
Should not an assertion be added there (then `when:` attribute of tasks below could be removed) ?
```
assert:
  that:
    file_attributes_result|changed
```

Reply:
No, the file module is [designed to fail](https://github.com/tchernomax/ansible/blob/devel/lib/ansible/module_utils/basic.py#L1282) if the extended attribute can't be applied.
On the contrary the get_file_attributes is [designed not to fail](https://github.com/tchernomax/ansible/blob/devel/lib/ansible/module_utils/basic.py#L1299) if `lsattr` isn't present or plate-forme doesn't support extended attributes.

If I where to replate `ignore_errors: True` with what you propose, the test would fail on all the plate-forme which doesn't support `chattr` or the 'A' extended attributes.

What I want to test is: **if** the plate-forme support the 'A' extended attribute **and** this attribute is already set ; the file module should **not** change the file.